docs/misc/commands.md

```diff
@@ -0,0 +1,211 @@
+                    player.sendServerMessage(Component.literal("You're now on the level: " + 
+                        StringArgumentType.getString(context, "level"))); // And we send a message to the executor
+                }
+                return Command.SINGLE_SUCCESS;
+            })
+    );
+}
+```
+:::warning
+**Do not use special characters or spaces** in `Commands.argument` as the name of the argument or in `Commands.literal` as the name of the command. It won't make the game crash, but it will make the command unusable and always give errors.
+:::
+:::tip
+You can create optional argument like this:
+```java
+dispatcher.register( // Assuming we already have a variable holding this dispatcher
+    Commands.literal("mycommandwithoptionalargs")
+    .then(Commands.argument("level", StringArgumentType.word())
+    .then(Commands.argument("amount", IntegerArgumentType.integer()))
+    .executes(/*execution with two arguments*/)) // The executes block is inside the then block
+    .executes(/*execution with one argument*/) // The executes block is after the then block
+);
+```
+:::
+## Creating argument types
+Sometime creating a `StringArgumentType` or `ResourceOrTagKeyArgument` isn't pratice or useful.
+In this case you can create your own.
+
+To create an argument type, you'll need to create a new class that implements `ArgumentType<T>` and its functions.
+```java
+public class CarArgumentType implements ArgumentType<Car> { // We create argument type based on the class Car
+    public Car parse(StringReader reader) { // To parse the written string to a car instance
+        return CARS.get(ResourceLocation.read(reader)); // If we have a HashMap<ResourceLocation, Car>
+    }
+    public <S> CompletableFuture<Suggestion> listSuggestion(CommandContext<S> ctx, SuggestionBuilder builder) { // To suggest all the different cars
+        return SharedSuggestionProvider.suggest(CARS.keySet().map(key::toString), builder)
+    }
+    public Collection<String> getExamples() { // To provide examples shown in the chat on wrong command writing.
+        return List.of("Volvo", "Citroën");
+    }
+}
+```
+
+You will also need to add some functions to allow instantiation and get argument:
+ - A `public static CarArgumentType car()` that returns a `new CarArgumentType()` (it's a best pratice to avoid direct instanciations).
+ - A function to get our arguments:
+ ```java
+ public static Car getCar(CommandContext<CommandSourceStack> ctx, String name) {
+    return ctx.getArgument(name, Car.class);
+ }
+ ```
+
+:::tip
+If you just wanna have custom suggestions like for an action (add, set, sub...) you can directly use `#suggests` after a `then` (that use a `RequiredArgumentType`) with as argument an equivalent of `CarAgumentType::listSuggestion`
+Another options is to create an enum with all the possibilities and use `EnumArgument.enumArgument(Cars.class)`.
+:::
+
+### `ArgumentTypeInfo`
+This class allow to serialize argument and provide build context.
```

Comment:
The point of the info is to sync any data that is stored when constructing the argument type to the client (only for server commands).